src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1061,8 +1073,9 @@ VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesInRange(
   BatchDecodedOutput output(numOutputFrames, options, streamMetadata);
 
   for (int64_t i = start, f = 0; i < stop; i += step, ++f) {
-    DecodedOutput singleOut = getFrameAtIndex(streamIndex, i);
-    output.frames[f] = singleOut.frame;
+    auto preAllocatedOutputTensor = output.frames[f];
+    DecodedOutput singleOut =
+        getFrameAtIndex(streamIndex, i, preAllocatedOutputTensor);
```

Comment:
Since we're not using pointers, we can make this even simpler than before:
```
DecodedOutput singleOut =
        getFrameAtIndex(streamIndex, i, output.frames[f]);
```
That is, everything should work fine and be clear without a local alias.